mne/preprocessing/_css.py

```diff
@@ -0,0 +1,115 @@
+# Author: John Samuelsson <johnsam@mit.edu>
+
+import numpy as np
+from ..utils import logger
+from ..evoked import EvokedArray, Evoked
+from .. import pick_types
+from numpy import linalg
+
+
+def temp_proj(ref_2, ref_1, raw_data, n_proj=6):
+    """Removes common signal subspace of ref_2 and ref_1 from raw_data.
+
+    Parameters
+    ----------
+    ref_2 : np.ndarray of float, shape (n_sensors_2, n_times)
+        The magnetometer data for CSS. Can use either all magnetometer data or
+        a few selected sensors close to a region to be suppressed.
+    ref_1 : np.ndarray of float, shape (n_sensors_1, n_times)
+        The gradiometer data for CSS. Can use either all gradiometer data or
+        a few selected sensors close to a region to be suppressed.
+    raw_data : np.ndarray of float, shape (n_sensors_raw, n_times)
+        The data to be filtered, typically the EEG data.
+    n_proj : int
+        The number of projection vectors.
+
+    Returns
+    -------
+    filtered_data : np.ndarray of float, shape (n_sensors_raw, n_times)
+        The filtered data.
+
+    Notes
+    -------
+    This temporal projection procedure removes the common signal subspace
+    between ref_2 and ref_1 from raw_data using n_proj number of
+    projection vectors. Normally used for cortical signal suppression, where
+    ref_1 is gradiometer data, ref_2 is magnetometer data and
+    raw_data is EEG data.
+
+    """
+    # Orthonormalize gradiometer and magnetometer data by a QR decomposition
+    Qg = linalg.qr(ref_1.T)[0]
+    Qm = linalg.qr(ref_2.T)[0]
+
+    # Calculate cross-correlation
+    C = np.dot(Qg.T, Qm)
+
+    # Channel weights for common temporal subspace by SVD of cross-correlation
+    Y, S, Z = linalg.svd(C)
```

Comment:
single- and even two-character variable names are discouraged (with exceptions sometimes made for iteration variables). Also, [we use snake case](https://mne.tools/dev/install/contributing.html#use-consistent-variable-naming) for variables, camel case for classes, and all caps for constants.  This would suggest names like `q_mag`, `q_grad`, `cross_corr`, etc

Reply:
harder to grep and replace :)